### PR TITLE
WIP: investigating SI-8069

### PR DESCRIPTION
**Pull request by adriaanm**

### CODE REVIEW

File: src/library/scala/concurrent/Future.scala [hunk 1]
```diff
@@ -649,8 +649,8 @@ object Future {
                     // up to the invoking executor
                     val remaining = _tasksLocal.get
                     _tasksLocal set Nil
-                    unbatchedExecute(new Batch(remaining)) //TODO what if this submission fails?
-                    throw t // rethrow
+                    try unbatchedExecute(new Batch(remaining)) // investigating SI-8069
```
Comment (viktorklang) [inline src/library/scala/concurrent/Future.scala:652]: LGTM, thanks @adriaanm

**Comment by gkossakowski** [comment]
Shall we merge this?

**Comment by adriaanm** [comment]
I never got confirmation this fixed the underlying support ticket. Should at least remove "WIP" from the commit.

**Comment by viktorklang** [comment]
In any case, it needs a regression test.

**Comment by adriaanm** [comment]
I don't have one -- I suspected this masked another issue because it swallows the `t` exception.

**Comment by adriaanm** [comment]
2.10 is not high priority anymore, so unless I get prodded by the original reporter, I won't come back to this.

**Comment by viktorklang** [comment]
It seems like there's a possible regression test in the report: https://issues.scala-lang.org/browse/SI-8069

**Pull request closed without merging**